pkg/lint/lintersdb/custom_linters.go

```diff
@@ -115,8 +116,11 @@ func (m *Manager) lookupAnalyzerPlugin(plug *plugin.Plugin) ([]*analysis.Analyze
 		return nil, err
 	}
 
-	m.log.Warnf("plugin: 'AnalyzerPlugin' plugins are deprecated, please use the new plugin signature: " +
-		"https://golangci-lint.run/contributing/new-linters/#create-a-plugin")
+	// TODO(ldez): remove this env var in the next minor version (v1.55.0)
+	if os.Getenv("GOLANGCI_LINT_HIDE_WARNING_ABOUT_PLUGIN_API_DEPRECATION") != "1" {
+		m.log.Warnf("plugin: 'AnalyzerPlugin' plugins are deprecated, please use the new plugin signature: " +
+			"https://golangci-lint.run/contributing/new-linters/#create-a-plugin")
```

Comment:
Should we add something like this in the log?
> You can hide this message by setting `GOLANGCI_LINT_HIDE_WARNING_ABOUT_PLUGIN_API_DEPRECATION`

Reply:
For me it's not a real option, it's more of a hidden feature only for people that read the changelog :sweat_smile: